Generalize spinnaker test to work for all devops templates

ejizba opened the pull request:
The generic test:
1. Deploys the template
2. Runs a basic smoke test
3. Uploads correlation id to storage account

ejizba committed:
Rename to generic qs template test

ejizba commented:
Since git is sometimes bad at picking up file renames, I did the renames in one commit and all my other changes in a second commit. See just the second commit for a better diff for reviewing: https://github.com/azure-devops/ci/pull/15/commits/34a7f196e4992445fe370bff5d851c2d94749e36

clguiman commented on test-quickstart-template.groovy:
```diff
@@ -4,50 +4,42 @@ node('quickstart-template') {
-                withCredentials([string(credentialsId: 'QsStorageConnectionString', variable: 'connection_string')]) {
-                    sh script_path + ' -tn ' + env.JOB_BASE_NAME + ' -sn ' + scenario_name + ' -ai ' + env.app_id + ' -ak ' + env.app_key + ' -cs "' + env.connection_string + '"'
+        if (run_tests) {
+            sh 'ssh -F ' + scenario_name + '/ssh_config -f -N tunnel-start'
```
can you run this outside of a stage?

ejizba replied:
I believe so. I'm running Jenkins jobs on my own instance right now to verify these changes and they're not to this point yet

ejizba replied:
I confirmed this runs just fine outside of a stage.

ejizba committed:
Generalize spinnaker test to work for all devops templates

The generic test:
1. Deploys the template
2. Runs a basic smoke test
3. Uploads correlation id to storage account
